### PR TITLE
Fix failing e2e tests and some refactoring

### PR DESCRIPTION
**Pull request by siemiatj**
Fix failing tests (mostly picking) and also optimize the DocumentList lifecycle getting rid of unnecessary data requests that are causing the UI to reload for no reason.

Related to #10177

### CODE REVIEW

File: frontend/src/__tests__/actions/FiltersActions.test.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ describe('FiltersActions general', () => {
     );
   });
 
-  it(`dispatches 'UPDATE_ACTIVE_FILTER' action when creating the filters`, () => {
+  it.only(`dispatches 'UPDATE_ACTIVE_FILTERS' action when creating the filters`, () => {
```
Comment (petrican) [inline frontend/src/__tests__/actions/FiltersActions.test.js:72]: Is this on purpose or you forgot it in here ?

**Review by petrican — CHANGES_REQUESTED** [review]

**Comment by petrican** [comment]
Screenshot 2020-11-06 at 08 27 37
Fetched the branch locally and ran the tests. For some reason there are 2 failing. I saw that the previous build failed due to similar reason.

**Comment by petrican** [comment]
Ran this morning e2e against dev1337 

Screenshot 2020-11-06 at 08 54 52

It seems there still are some picking tests failing. https://jenkins.metasfresh.com/job/ops/job/run_e2e_tests/1044/

https://dashboard.cypress.io/projects/5yp4q1/runs/902/specs

**Comment by petrican** [comment]
Just checked again and I see the unit tests are passing
Screenshot 2020-11-06 at 09 49 18
My local branch was not in sync

**Comment by siemiatj** [comment]
> Ran this morning e2e against dev1337
> 
> Screenshot 2020-11-06 at 08 54 52
> 
> It seems there still are some picking tests failing. https://jenkins.metasfresh.com/job/ops/job/run_e2e_tests/1044/
> 
> https://dashboard.cypress.io/projects/5yp4q1/runs/902/specs

Yeah I know. They passed once and now are failing every single time again even though they're running fine locally.

**Pull request merged**